src/python/gudhi/off_utils.pyx

```diff
@@ -25,21 +26,96 @@ __license__ = "MIT"
 
+def _get_next_line(file_desc, comment='#'):
+    """Return the next line that is not a comment.
+
+    :param file_desc: An open file in read mode.
+    :type file_desc: file
+    :param comment: The characters or list of characters used to indicate the start of a comment.
+    :type comment: string
+
+    :returns: The next line.
+    :rtype: string
+    """
+    while True:
+        # file_desc.readline() is preferred to next(file_desc), as the second option is not compatible with
+        # seek and tell methods
+        line = file_desc.readline()
+        if not line.startswith(comment):
+            break
+    return line
+
+def _read_off_file_header(file_desc):
+    """Return the information contained in the header of an OFF file.
+
+    :param file_desc: An open file in read mode.
+    :type file_desc: file
+
+    :returns: The point cloud dimension, and the number of points.
+    :rtype: tuple(int, int)
+
+    Raises:
+        ValueError: If the file does not respect the OFF file format.
+    """
+    nb_vertices = -1
+
+    line = _get_next_line(file_desc)
+    # First line should be "OFF" (3d case) or "nOFF" (dD case)
+    if line.lower().startswith("off"):
+        dim = 3
+    elif line.lower().startswith("4off"):
+        dim = 4
+    elif line.lower().startswith("noff"):
+        # "nOFF" case, next line is the dimension
+        # can also contain nb_vertices, nb_faces nb_edges (can also be on the next line)
+        line = _get_next_line(file_desc)
+        digits = [int(s) for s in line.split() if s.isdigit()]
+        dim = digits[0]
+        if len(digits) > 1:
+            nb_vertices = digits[1]
+            # nb_faces =  digits[2]
+            # nb_edges =  digits[3] # not used - can be ignored
+            # nb_cells =  digits[4]
+    else:
+        raise ValueError(f"Inconsistent OFF header, got '{line.rstrip()}', should be 'OFF', '4OFF' or 'nOFF'")
+        
+    # nb_vertices can be already set by "nOFF" case, when 'dim nb_vertices nb_faces nb_edges' on the same line
+    if nb_vertices < 0:
+        # Number of points is the first number ("OFF" case) or the second one ("nOFF" case) of the second line
+        line = _get_next_line(file_desc)
+        digits = [int(s) for s in line.split() if s.isdigit()]
+        nb_vertices = digits[0]
+        # nb_faces =  digits[1]
+        # nb_edges =  digits[2] # not used - can be ignored
+        # nb_cells =  digits[3]
+    # If comments between header and the first point
+    line = _get_next_line(file_desc)
+    # Here the first line without comment is read - let's go back to the beginning of this line
+    file_desc.seek(file_desc.tell() - len(line))
+    return dim, nb_vertices
+
 def read_points_from_off_file(off_file=''):
     """Read points from an `OFF file <fileformats.html#off-file-format>`_.
 
     :param off_file: An OFF file style name.
     :type off_file: string
 
     :returns:  The point set.
```

Comment:
(not for this PR) I wonder if it would make sense to have an option to return the opened file, so the caller can keep reading simplices from the right place. Maybe reading simplices really belongs in a different function though.

Reply:
Yes, I keep in mind some ideas to read vertices coordinates and faces from OFF files. This could be used at least by the extended persistence notebook